includes/class-wc-checkout.php

```diff
@@ -980,6 +980,8 @@ protected function process_order_payment( $order_id, $payment_method ) {
 			}
 
 			wp_send_json( $result );
+		}else{
+			do_action( 'woocommerce_payment_failed', $result, $order_id );
```

Comment:
Let's be sure to [document](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/) this new hook.